### PR TITLE
v2: kanban, timeline, detail view, stats, column renderers, hidden filters, top actions

### PR DESCRIPTION
**Pull request by hadifarnoud**
Substantial additive release. All changes are **backwards-compatible** — no existing controller, view, or test stops working. Regression-tested against a downstream app (~43 tests pass when this branch is overlaid into the app's `vendor/`).

### New view variants

- **Detail view** ([docs](docs/detail-view.md)) — opt-in `show()` page with sections + sidebar + timeline. Controllers register cards/panels in `setup()`; falls back to the existing read-only-form rendering when not registered.
- **Kanban view** ([docs](docs/kanban.md)) — drop-in column view gated on `?view=kanban`. `KanbanConfig` value object declares columns, card shape, and the transition route; SortableJS wiring left to the app via two exposed JS globals.
- **Timeline** ([docs](docs/timeline.md)) — `addTimelineSource(name, producer)` merges multiple event streams (transitions, audits, comments) into one sorted feed. New `Timeline` + `TimelineEvent` value objects.
- **Stats** ([docs](docs/stats.md)) — `addStat(label, fn, options)` for summary-number cards above the list/kanban. Compute closures are lazy.

### Controller API additions

- `addHiddenFilter(input, callback)` — filters that apply when their query param is present but render no UI widget. Useful for deep links.
- `addSearchField(columns[], input, field?)` — multi-column LIKE search with `%`/`_` escaping. Replaces hand-rolled `whereRaw` patterns.
- `addTopAction(caption, route, options)` — page-header buttons (not per-row).
- `setShowView(view)` / `addDetailSection` / `addDetailSidebar` — detail-view registration.
- `addTimelineSource(name, producer)` / `getTimeline()` — timeline registration.
- `enableKanban(KanbanConfig)` / `getKanbanConfig()` — kanban registration.
- `addStat(label, compute, options)` / `getStats()` — stats registration.

### Field flags

- `Field::readOnly()` / `isReadOnly()` — `Form::saveToModel` skips read-only fields entirely, so they render in the form but never accept request input. Useful for service-managed attributes.
- `Field::showWhen(Closure)` / `shouldShow($model)` — predicate-based conditional visibility.

### Column renderers

- New `Kamva\Crud\Columns\Renderers` helper namespace:
  - `Renderers::badge($attr, $opts)` — Bootstrap badge
  - `Renderers::link($attr, $hrefFn, $opts)` — `<a>` element
  - `Renderers::boolean($attr, $opts)` — check / dash
  - `Renderers::truncate($attr, $limit)` — shortened text with ellipsis
  - `Renderers::date($attr, $format)` — formatted date
- Each returns a `Closure` that drops into the existing `addColumn($title, $value)` API. All escape user input.

### Quick wins

- `FilterContainer::render()` returns `''` instead of throwing when no field is attached. `hasField()` and `getField()` added for views.
- `index()` now pre-filters hidden filters out of `$filters` so existing list templates that iterate `\$filter->render()` don't render empty divs.
- `index()` passes `$topActions` and `$stats` to list/kanban views.

## Documentation

First README for the package (the repo previously had no README at all). Plus per-feature docs in `docs/`:

| File | Covers |
|---|---|
| `README.md` | Overview, install, quick start, feature index |
| `docs/fields.md` | Fields including v2 `readOnly()` + `showWhen()` |
| `docs/columns.md` | Columns + v2 `Renderers` helpers |
| `docs/filters.md` | Filters incl. `addHiddenFilter` + `addSearchField` |
| `docs/actions.md` | Row actions + v2 `addTopAction` |
| `docs/detail-view.md` | New |
| `docs/timeline.md` | New |
| `docs/kanban.md` | New, with full JS wiring example |
| `docs/stats.md` | New |

## Tests

First test suite for the package. `orchestra/testbench` + PHPUnit 11. **50 unit tests / 102 assertions** cover:

- `FilterContainer` null-safety
- `BaseField` `readOnly()` + `showWhen()` flags
- Controller API additions
- `Columns\Renderers` escaping + edge cases
- `Timeline` source merging + sorting + malformed-entry skipping
- `KanbanConfig` bucketing + enum resolution + value_sum + id fallback
- `Stats` lazy compute

Run with `composer test`. `tests/Stubs/Controller.php` provides the `App\Http\Controllers\Controller` base class that `CRUDController` extends (a Laravel-app convention the package assumes).

## Backwards compatibility

Two soft-behaviour changes worth flagging during review:

1. `FilterContainer::render()` now returns `''` instead of throwing when no field is attached. Any existing user-published list view that relied on the throw (very unlikely) would now render an empty `<div>`. The package's `index()` pre-filters hidden filters out of `\$filters` before passing to the view, so default behaviour is unchanged.

2. `show()` now dispatches to the detail view IF the controller called `addDetailSection` / `addDetailSidebar` / `addTimelineSource` / `setShowView`. Controllers that don't touch these methods continue to render the read-only edit form as before.

Everything else is purely additive: new methods, new classes, new view templates that didn't exist before.

## Discovery context

This work originated while building a sales-funnel CRM feature in a downstream Laravel app. The downstream app's pain points (custom kanban, custom detail page, custom timeline) suggested these primitives belong in the framework. The downstream app keeps working unchanged; once this PR lands, it can migrate piece by piece to the new APIs.

### CODE REVIEW

File: src/Fields/Internal/BaseField.php [hunk 1]
```diff
@@ -286,4 +286,62 @@ public function shouldSkipSaving(): bool
...
+     *
+     * Has no effect on saving — combine with {@see readOnly()} if you also
+     * want to reject writes when the field is hidden.
+     *
+     * @return $this
+     */
+    public function showWhen(\Closure $predicate): FieldContract
+    {
+        $this->setConfig('showWhen', $predicate);
+        return $this;
```
[the remark this answers is not in the document]
Comment (hadifarnoud) [reply src/Fields/Internal/BaseField.php:332]: Fixed in 65254b4. The render-time enforcement is up to view templates (the package's stock views are stubs — apps publish + implement them), but the load-bearing server-side enforcement is now wired:

- `Form::saveToModel` now filters out fields whose `shouldShow($model)` returns false alongside the existing readOnly/skip checks. This prevents the UI-bypass attack where someone crafts a request including a value for a hidden field.
- New `Form::getVisibleFields($model)` helper that returns only fields passing the predicate. The controller now passes $visibleFields to the create/edit view, so apps that want render-time filtering iterate it instead of $form->getFields().

Tests: `tests/Unit/FormSaveTest.php` covers all the cases (readOnly + showWhen filtering on save, getVisibleFields edge cases including null model, conditional predicates that depend on existing model state).

File: src/Timeline/Timeline.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    {
+        $events = collect();
+
+        foreach ($this->sources as $producer) {
+            $items = $producer($model) ?? [];
+            foreach ($items as $item) {
+                if ($item instanceof TimelineEvent) {
+                    $events->push($item);
+                } elseif (is_array($item)) {
+                    $events->push(TimelineEvent::fromArray($item));
```
[the remark this answers is not in the document]
Comment (hadifarnoud) [reply src/Timeline/Timeline.php]: Fixed in 65254b4. `Timeline::for()` now wraps each entry's normalisation in try/catch — a bad `at` string (or any other DateTimeImmutable parse failure) skips just that entry instead of taking down the whole detail page. Also wrapped the source-producer invocation in try/catch so a single source throwing doesn't break sibling sources.

Tests:
- `test_array_entry_with_bad_at_string_is_skipped_not_thrown` — verifies one bad `'at'` string in a producer's output doesn't bubble up
- `test_throwing_producer_does_not_break_other_sources` — verifies one exploding source still lets the others render

File: src/CRUDController.php [hunk 3]
```diff
@@ -399,8 +405,28 @@ public function index(Request $request)
...
+                $columns = $this->kanban->bucket($rows->get());
+                return view('kamva-crud::kanban', [
+                    'title'           => $title,
+                    'createRoute'     => $createRoute,
+                    'filters'         => $filters,
+                    'topActions'      => $topActions,
+                    'stats'           => $stats,
+                    'columns'         => $columns,
+                    'kanban'          => $this->kanban,
+                    'transitionUrl'   => route($this->kanban->transitionRoute, ['__ID__'] + $this->routeParameters),
```
[the remark this answers is not in the document]
Comment (hadifarnoud) [reply src/CRUDController.php]: Fixed in 65254b4. Swapped `['__ID__'] + $this->routeParameters` (PHP array-union, drops positional keys) for `array_merge($this->routeParameters, ['__ID__'])` which preserves named string keys AND re-indexes positional keys so they land left-to-right in the right route segments.

Tests: `tests/Unit/Kanban/KanbanTransitionUrlTest.php` covers three regression cases against real registered Laravel routes:
- Named parent params (`['organization' => 'abc']` + flat `{lead}` placeholder) — produces `/organization/abc/lead/__ID__/transition`
- No parent params — produces `/lead/__ID__/transition`
- Positional parent param — still drops the placeholder into the correct segment thanks to array_merge re-indexing
